### ISSUE TITLE
[BUG] Harvester VMs restored from a backup are not editable due to missing volume information.

### ISSUE THREAD

**Issue by zanewgray** [harvester/harvester#2634]
**Describe the bug**
Harvester VMs restored from a backup are not editable due to missing volume information.

**To Reproduce**
1. Create an S3 backup target
2. Create a new VM with 2vCPU, 8GB memory, a volume "disk-1" (size doesn't matter), and an appropriate image (K3OS in this case)
4. Backup the running VM to the S3 target
5. Stop the VM
6. Restore the VM from the backup on the S3 target
7. Stop the VM
8. Edit the VM config and change the vCPU from "2" to "4"
9. Attempt to save the VM config
10. Note the error regarding "volume disk-1: 'Volume' is required!"
11. Navigate to the Volumes for the VM.  Note that the Volume section for disk-1 is empty.

**Expected behavior**
Restored VMs should be editable and the Volume entry for the VM disk should not be blank.  The VM volume should reflect the backup volume (or similar)

**Support bundle**
[too large to attach to support case - 65MB in size, but GitHub has a 25MB limit]  

Uploaded here:
https://sooners-my.sharepoint.com/:u:/g/personal/zgray_ou_edu/EcBs3z6yZ7pIpCZJddpoorAB-I7wPnbcPEUf7pXwBI9UhQ?e=9OtlYq

**Environment**
 - Harvester ISO version: 1.0.2
 - Underlying Infrastructure 3 each bare metal Dell PowerEdge R740XDs

**Additional context**

**Comment by n313893254** [comment]
Hi @zanewgray, I can not reproduce the issue, are you reproduce it always? If so, could you help check `spec.template.spec.volumes.persistentVolumeClaim.claimName` of VM whether can be found in PVC

**Comment by zanewgray** [comment]
@n313893254 after posting this bug report on Friday, I did check the spec.template.spec.volumes.persistentVolumeClaim.claimName under the "Edit YAML" for the VM.  Here is where things got odd.  Sometimes opening the YAML would show a value of "null" for the claimName.  Then, closing and reopening the YAML for the VM might show the real volume under the claimName, but not always. 

I managed to resolve my issue by opening and closing the YAML definition for the VM until the proper claimName was displayed for the volume (as opposed to a "null" value).  If a proper volume was listed, then I could successfully edit the vCPU for the VM and save the changes.  Using this method I was able to successfully make my changes and boot the VMs.

At no time did the GUI editor for the VM show the proper volume name.  It would always appear blank in the VM Volumes section of the GUI editor with no option to select a volume.

One item of note: these particular VMs (Rancher K3OS instances) have been restored multiple times from different backup versions (meaning that the claimName has changed multiple times over the life of the VMs).  I am unsure if this might have some impact on the reproducibility.

Thanks for the assistance!
